docs/table_design/list_partitioning.md

```diff
@@ -0,0 +1,110 @@
+# List partitioning
+
+Since v3.1, StarRocks supports List partitioning. Data is partitioned based on a predefined value list  for each partition, which can accelerate queries and facilitate management according to enumerated values.
+
+## Introduction
+
+You need to explicitly specify the column values list in each partition. These values do not need to be continuous, unlike the continuous time or numeric range required in Range Partitioning. During data loading, StarRocks will store the data in the corresponding partition based on the mapping between the data's partition column values and the predefined column values for each partition.
```

Comment:
![image](https://github.com/StarRocks/starrocks/assets/57167462/83a138fe-8c9b-49d1-8a3c-e053f3cfc2bb)
I think an agreement about how to write these partitioning methods should be reached. For example, List Partitioning, Range Partitioning, and Expression Partitioning, if we treat them as proper nouns; or list partitioning, range partitioning, and expression partitioning, if we do not treat them as proper nouns. Either is fine. But we need to adhere to one style.